ui/scripts/vpc.js

```diff
@@ -745,12 +745,24 @@
                         }
                     },
                     dataProvider: function(args) {
+                        var array1 = [];
+                        if (args.filterBy != null) {
+                            if (args.filterBy.search != null && args.filterBy.search.by != null && args.filterBy.search.value != null) {
+                                if (args.filterBy.search.by === "name") {
+                                    if (args.filterBy.search.value.length > 0)
+                                        array1.push("&keyword=" + args.filterBy.search.value);
+                                }
+                            }
+                        }
+                        var data = {
+                            page: args.page,
+                            pageSize: pageSize,
+                            networkid: args.context.networks[0].id,
+                            listAll: true
+                        };
                         $.ajax({
-                            url: createURL('listLoadBalancers'),
-                            data: {
-                                networkid: args.context.networks[0].id,
-                                listAll: true
-                            },
+                            url: createURL('listLoadBalancers' + array1.join("")),
+                            data: data,
                             success: function(json) {
```

Comment:
@vedulasantosh much better, however, that's not why I meant. It still does not remove the array/join usage. On line 764, remove appending of `array1.join()` usage, instead move the `data` object after line 747, and at line 753, push the key/value to the `data` object.